src/main/java/me/karakelley/http/handlers/StaticFilesHandler.java

```diff
@@ -53,13 +53,17 @@ private Response serveFilesResponse(Response response, String requestedResource)
     return response;
   }
 
-  private ArrayList<String> getFilesAndDirectoryLinks(String resource) {
+  private String getFilesAndDirectoryLinks(String resource) {
     List<File> files = publicDirectory.getDirectoriesAndFiles(resource);
     ArrayList<String> resources = new ArrayList<>();
     for (File file : files) {
-      resources.add(contentPresenter.displayResources(file, publicDirectory));
+      resources.add(createLinks(file));
     }
-    return resources;
+    return contentPresenter.displayResources(String.join("", resources));
+  }
+
+  private String createLinks(File fileName) {
```

Comment:
Why did this leave the `HtmlPresenter`? Seems pretty HTML-specific to me...

Reply:
I went back and forth on this. This is something I struggle with sometimes when I have a class implementing an interface like `ContentPresenter` and I need a new specific function for an implementing class.  It doesn't make sense for `ContentPresenter` to have a function `createLinks` because that wouldn't make sense if other content types were introduced like JSON. But it also feels a little weird for it to be here considering there's an `HtmlPresenter` class.

Reply:
Let's chat about this tomorrow (Friday).